src/pept-compare/protein.py

```diff
@@ -5,6 +5,14 @@
 from methods import *
 from pyteomics import parser
 import numpy as np
+from joblib import Memory
+
+memory = Memory(".cache/", verbose = False)
+
+@memory.cache
+def download_fasta(protein_id):
+    link = f"http://www.uniprot.org/uniprot/{protein_id}.fasta"
+    return requests.get(link).text
```

Comment:
This caches the protein data meaning that you only have to download it once. The data is stored using the python-pickle protocol in `./cache`.